zerver/lib/create_user.py

```diff
@@ -127,4 +132,23 @@ def create_user(email: str, password: Optional[str], realm: Realm,
     user_profile.save(update_fields=["recipient"])
 
     Subscription.objects.create(user_profile=user_profile, recipient=recipient)
+
+    if user_avatar_url:
+        from zerver.lib.upload import upload_avatar_image_from_url
+        from zerver.lib.actions import do_change_avatar_fields
+
+        try:
+            avatar_hash = upload_avatar_image_from_url(user_avatar_url, user_profile)
+            do_change_avatar_fields(user_profile, UserProfile.AVATAR_FROM_USER)
+            user_profile.avatar_hash = avatar_hash
+            user_profile.save(update_fields=["avatar_hash"])
+
+        except Exception:
```

Comment:
I'm not sure if we should be catching all exceptions, or some specific ones that can come from download/upload failures? :thinking:  Either way, we should at least log what the exception is, in the `warning` below.